contracts/utils/MultiRateLimited.sol

```diff
@@ -0,0 +1,273 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {CoreRef} from "../refs/CoreRef.sol";
+import {TribeRoles} from "./../core/TribeRoles.sol";
+import {RateLimited} from "./RateLimited.sol";
+import {IMultiRateLimited} from "./IMultiRateLimited.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+/// @title contract for putting a rate limit on how fast an address can perform an action e.g. Minting
+/// there are two buffers, one buffer which is each individual addresses's current buffer,
+/// and then there is a global buffer which is the buffer that each individual address must respect as well
+/// @author Fei Protocol
+contract MultiRateLimited is RateLimited, IMultiRateLimited {
+    using SafeCast for *;
+
+    /// @notice the struct containing all information per rate limited address
+    struct RateLimitData {
+        uint32 lastBufferUsedTime;
```

Comment:
Nit: if we change this to uint112 we'll pack the two slots perfectly. Right now the first slot is only using 176/255 bits.

Reply:
perfect is the enemy of the good

Reply:
catchphrases are the enemy of communication efficiency ;)

Reply:
(closing, will fix in my nitpicks pr)